splat.pl

```diff
@@ -0,0 +1,953 @@
+
+    if (substr ($aca_packet, 3, 40) eq "\377"x40) {  # PEA off
+        $pea_on = 0;
+	for $slot (0..7) {
+	    my $update_slot = 0;
+	    for $i (0..7) {
+		for $j (0..7) {
+		    $update_slot = 1 if ($img_array[$slot][$i][$j] != 0.0);
+		    $img_array[$slot][$i][$j] = 0.0;
+		}
+	    }
+	    draw_slot($slot);
+	}
+	return;
+    }
+
+    $pea_on = 1;
+    $startbit = 40;		# start of "image types" in aca_packet_bits
+    for $slot (0 .. 7) {
+#       next unless (extr_val ($aca_packet_vec, 68+$slot*216, 2));    # idle slot?
+	$img_type = extract ($aca_packet_vec, 3);
+	$cal{IMGTYPE}[$slot] = $img_type;
+
+	# Put header data into $img_data[$slot] if needed
+	if ($img_type == 0 || $img_type == 1 || $img_type == 4) {
+	    $img_data[$slot] = substr $aca_packet, 0, 5;
+	}
+
+	# identify a 6x6 or 8x8 readout slot, for printing out temperatures
+	if ($img_type == 2 || $img_type == 5) {$slot68 = $slot};
+
+	# Copy data from the aca_packet (224 bytes) to img_data[slot]
+	$img_data[$slot] .= substr $aca_packet, $slot*27+8, 27;
+
+	# Decom data in $img_data[$slot] if needed
+	if ($img_type == 0 || $img_type == 2 || $img_type == 7) {
+	    $len = $img_data_length[$img_type];
+	    $lenbuf = 27*4+5;	# length of $img_data_vec in bytes
+	    if (length $img_data[$slot] < $lenbuf) {
+		$img_data[$slot] .= "\000" x ($lenbuf - length $img_data[$slot]);
+	    }
+
+	    $img_data_vec->Block_Store(scalar reverse $img_data[$slot]);
+
+	    # Extract the data
+
+	    @{$fields[$slot]} = ();
+	    for $i (0 .. $#d_field) {
+		$img_data_vec_startbit = $d_byte[$i] * 8 + $d_bit[$i];
+		last if ($img_data_vec_startbit + $d_len[$i] > $len * 8);
+		push @{$fields[$slot]}, $d_field[$i];
+		$raw{$d_field[$i]}[$slot] =
+		    extr_val ($img_data_vec, $img_data_vec_startbit, $d_len[$i]);
+	    }
+
+	    apply_cal ($slot);
+	    set_image_array ($slot, $img_type);
+	    write_slot_header($img_type, $slot)
+		if (defined $opt_out
+                        && defined $DAT_fh[$slot]
+                            && not defined $WROTE_HEADER[$slot]
+                    );
+	    write_slot_data($img_type, $slot)
+		if (defined $opt_out && defined $DAT_fh[$slot]);
+	}
+    }
+    $top->update();
+}    
+
+
+##***************************************************************************
+sub write_slot_header {
+##***************************************************************************
+# write slot data to a file, for the specified slot
+# if temperature etc. data is not available, put out dummy data = -99.
+    my $img_type = shift;
+    my $slot = shift;
+
+    $dutc = $opt_tctm ? $utc1970 : -99;
+    print {$DAT_fh[$slot]} "VCDU"," ","time"," ","DUTC"," ","SLOT"," ","IMGTYPE"," ";
+    for $i (0 .. 11) {
+	print {$DAT_fh[$slot]} $fields[$slot][$i] , " ";
+    }
+    if (defined $fields[$slot][36]) {
+	for $i (31 .. 36) {
+	    print {$DAT_fh[$slot]} $fields[$slot][$i], " ";
+	}
+    }
+    else {
+	print {$DAT_fh[$slot]} "-99 -99 -99 -99 -99 -99 ";
```

Comment:
I should just write in these values from a string list here instead of an if/else.